src/main/kotlin/no/nav/familie/ef/sak/repository/domain/Vedtaksbrev.kt

```diff
@@ -0,0 +1,14 @@
+package no.nav.familie.ef.sak.repository.domain
+
+
+import no.nav.familie.ef.sak.api.dto.BrevRequest
+import no.nav.familie.ef.sak.service.steg.StegType
+import org.springframework.data.annotation.Id
+import java.util.*
+
+data class Vedtaksbrev(@Id
+                       val id: UUID = UUID.randomUUID(),
+                       val behandlingId: UUID,
+                       val steg: StegType,
```

Comment:
Litt usikker på hvorfor vi har stegtype i vedtaksbrevet? Er dette en duplisering av samme steg som ligger i behandlingen?

Reply:
Har steg for å kunne si om det er brevet som er laget før eller etter totrinnskontroll. Tanken var at hver behandling har et  vedtaksbrev for hvert av de to stegene (send til beslutter og beslutte vedtak). Foreløpig er det bare en duplisering av behandlingssteget, på opprettelsestidspunktet for brevet. Kanskje ikke optimalt å bruke samme steg som i behandling, men vi må vel uansett ha et felt som sier om vedtaksbrevet er fra før eller etter totrinns?

Reply:
Hvis vi trenger to entiteter per behandling så anbefaler jeg å innføre egen verdi: foreksempel: utkast / ferdigstilt eller boolsk verdi med ferdigstilt. Da kan det være nyttig med en unique constraint på behandlingsId og denne verdien også. 

Alternativt kan vi ha `val utkastBrevRequest, val brevRequest, val utkastPdf, val pdf` som fire felter i en klasse, dersom dette er tett knyttet sammen. 

PS: Jeg har ikke reflektert mye nå, bare kaster ut idéer for løsning som kan unngå å "speile" behandlingen sin stegtype.

Reply:
Forslaget ditt med egen verdi og constraint høres fornuftig ut. Men jeg har egentlig ingen god grunn til å ville ha to entiteter i dette caset. Eneste jeg bygger det på er "immuteability good, muteability bad"-dogmaet 🤔